llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -54916,6 +54952,11 @@ static SDValue combineVectorCompare(SDNode *N, SelectionDAG &DAG,
       return DAG.getConstant(0, DL, VT);
   }
 
+  const TargetLowering &TLI = DAG.getTargetLoweringInfo();
+  if (TLI.SimplifyDemandedBits(
+          SDValue(N, 0), APInt::getAllOnes(VT.getScalarSizeInBits()), DCI))
+    return SDValue(N, 0);
+
```

Comment:
Will this not get called else where?
Either way, this seems like it should be a seperate commit.